bokehjs/src/lib/models/annotations/band.ts

```diff
@@ -31,9 +31,11 @@ export class BandView extends AnnotationView {
 
   connect_signals(): void {
     super.connect_signals()
-    this.connect(this.model.source.streaming, () => this.set_data(this.model.source))
-    this.connect(this.model.source.patching, () => this.set_data(this.model.source))
-    this.connect(this.model.source.change, () => this.set_data(this.model.source))
+    const update = () => this.set_data(this.model.source)
+    this.connect(this.model.change, update)
+    this.connect(this.model.source.streaming, update)
+    this.connect(this.model.source.patching, update)
+    this.connect(this.model.source.change, update)
```

Comment:
This pattern is repeated in many places, and it just asks for `this.connect` to be able to receive an array of signals. This would also allow reuse of `new_slot` within `connect`. @mattpap What do you think?

Reply:
That would be useful in general. However, in this case, this code most likely will be removed soon, as I will be bringing all vectorizable (or source-driven) renderers under one implementation.

Reply:
Sounds awesome!